src/index.ts

```diff
@@ -352,25 +366,63 @@ export function startAnalysis<
+                                }),
+                                `--set hardump=${flowsOutputPath}`,
+                            ],
+                            {
+                                stdio: ['pipe', 'pipe', 'pipe', 'ipc'],
+                                env: { ...process.env, IPC_PIPE_FD: '3' },
+                                shell: true, // We have to set this, because mitmdump doesn’t accept options starting with '--' as options in non-shell mode.
+                            }
+                        ),
                         flowsOutputPath,
                     };
-                    await timeout(awaitProcessStart(mitmproxyState.proc, 'Proxy server listening'), {
+
+                    await timeout(awaitMitmproxyStatus(mitmproxyState.proc, 'running'), {
                         milliseconds: 30000,
+                    }).catch((e) => {
+                        if (e.name === 'TimeoutError')
+                            throw new TimeoutError('Starting mitmproxy failed after a timeout.');
+                        throw e;
                     });
                 },
                 stopTrafficCollection: async () => {
-                    killProcess(mitmproxyState?.proc);
-                    if (mitmproxyState?.flowsOutputPath && inProgressTrafficCollectionName) {
-                        const trafficDump = await readFile(mitmproxyState?.flowsOutputPath, 'utf-8');
-                        res.traffic[inProgressTrafficCollectionName] = trafficDump;
-                    }
-
-                    inProgressTrafficCollectionName = undefined;
-                    // eslint-disable-next-line require-atomic-updates
-                    mitmproxyState = undefined;
+                    if (!mitmproxyState?.proc) throw new Error('No traffic collection is running.');
+                    await Promise.all([
+                        awaitProcessClose(mitmproxyState.proc).then(async () => {
+                            if (mitmproxyState?.flowsOutputPath && inProgressTrafficCollectionName) {
+                                try {
+                                    const trafficDump = await readFile(mitmproxyState?.flowsOutputPath, 'utf-8');
+                                    res.traffic[inProgressTrafficCollectionName] = JSON.parse(trafficDump);
+                                } catch {
+                                    throw new Error(
+                                        `Reading the flows from the temporary file @ "${mitmproxyState.flowsOutputPath}" failed.`
+                                    );
+                                }
+                            }
+
+                            inProgressTrafficCollectionName = undefined;
+                            // eslint-disable-next-line require-atomic-updates
+                            mitmproxyState = undefined;
+                        }),
+                        mitmproxyState.proc.kill(),
```

Comment:
Why are you not using `killProcess()` anymore?

Reply:
Ah, that is leftover from when I investigated a bug and though maybe killing the process only gracefully would solve it.